cegui/src/text/LegacyTextParser.cpp

```diff
@@ -257,13 +276,25 @@ void LegacyTextParser::processControlString(const std::u32string& ctrlStr, std::
 
     if (valueValid && key == ImageTagName)
     {
+        const String val = ctrlStr.substr(valueStart, valueEnd - valueStart);
+        if (!ImageManager::getSingleton().isDefined( val )) {
+            ImageManager::getSingleton().addBitmapImageFromFile( val, val );
+        }
+
         //!!!TODO TEXT: try to find the same image first!
 
-        auto element = std::make_unique<RenderedTextImage>(
-            PropertyHelper<Image*>::fromString(ctrlStr.substr(valueStart, valueEnd - valueStart)));
+        auto element = std::make_unique<RenderedTextImage>( PropertyHelper<Image*>::fromString(val) );
         element->setColour(d_colours);
         element->setBackgroundColour(d_bgColours);
-        element->setSize(d_imageSize);
+        element->setUseModulateColour(d_useModColour);
+        if (d_imageAspectLock) {
+            element->setSize(getRespectRatioSize(
+                d_imageSize,
+                ImageManager::getSingleton().get( val ).getRenderedSize()
```

Comment:
Redundant search in a manager. Consider accessing an image from an `element`, or better, making API like `element->setSize(d_imageSize, /*aspectLock=*/ true)`. This way you will not provide public access to internals.

Reply:
OK, I make `RenderedTextImage::setSize(const Sizef& size, bool aspectLock)` and move `getRespectRatioSize` to `RenderedTextImage.cpp`